rewrite-xml/src/main/java/org/openrewrite/xml/XmlParser.java

```diff
@@ -64,6 +67,11 @@ public Stream<SourceFile> parseInputs(Iterable<Input> sourceFiles, @Nullable Pat
         });
     }
 
+    private String preprocessJspSyntax(String source) {
+        // Replace JSP comments with XML comments to avoid parser errors
+        return source.replaceAll("<%--(.*?)--%>", "<!--$1-->");
+    }
```

Comment:
This would likely break print equality. Let's add a test first before continuing any further.

Reply:
So you want a unit test here 
Ok I will do it @timtebeek 
thanks

Reply:
Important to keep in mind here in the parsing process is that you can actually not change the code. 
If you would change the input code, the parsed code differs from whatever is really in the codebase. 
You would basically be running recipes against a codebase that never existed.

Rather than replacing these tags, I would expect some logic to parse them correctly. 
Jsp comments are "valid" code and could perfectly be parsed if we extend the XMLParser to understand other tags than the current one. 

For the unit test example: Have a look at my showcase of #4306 on [my fork](https://github.com/Jenson3210/rewrite/commit/d81cb5e62a594b25fe52dec171f2ea221d522cbe) I think you could start from this unit test, add a jsp comment and use that test as input for all parser fixes. 
This would then fix #4306 and #4240